docs/metasploit-framework.wiki/Attacking-ADCS-ESC-Vulnerabilities.md

```diff
@@ -0,0 +1,931 @@
+since most LDAP servers in an Active Directory (AD) environment are configured in such a way that they allow users to read most objects, but not write to them.
+For our purposes, since we just need to read the details of the certificate templates that are available, this means normal user permissions should be sufficient.
+
+To run the module, specify the login credentials for an AD user, and set `RHOSTS` to the address of one of the Domain Controller (DC) IP addresses, then enter `run`.
+This will cause the module to log into the LDAP server on the target DC, and list out the vulnerable certificate templates and which CA servers they are available from,
+as well as the permissions that are required to enroll in these certificate templates. The following is a sample output of running this against a test server:
+
+```
+msf6 > use auxiliary/gather/ldap_esc_vulnerable_cert_finder
+msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > show options
+
+Module options (auxiliary/gather/ldap_esc_vulnerable_cert_finder):
+
+   Name                  Current Setting  Required  Description
+   ----                  ---------------  --------  -----------
+   BASE_DN                                no        LDAP base DN if you already have it
+   DOMAIN                                 no        The domain to authenticate to
+   PASSWORD                               no        The password to authenticate with
+   REPORT_NONENROLLABLE  false            yes       Report nonenrollable certificate templates
+   RHOSTS                                 yes       The target host(s), see https://github.com/rapid7/metasploit
+                                                    -framework/wiki/Using-Metasploit
+   RPORT                 389              yes       The target port
+   SSL                   false            no        Enable SSL on the LDAP connection
+   USERNAME                               no        The username to authenticate with
+
+
+View the full module info with the info, or info -d command.
+
+msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > set DOMAIN DAFOREST
+DOMAIN => DAFOREST
+msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > set USERNAME normal
+USERNAME => normal
+msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > set PASSWORD normaluser
+PASSWORD => normaluser
+msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > set RHOSTS 172.30.239.85
+RHOSTS => 172.30.239.85
+msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > run
+[*] Running module against 172.30.239.85
+
+[*] Discovering base DN automatically
+[+] 172.30.239.85:389 Discovered base DN: DC=daforest,DC=com
+[*] Template: SubCA
+[*]    Distinguished Name: CN=SubCA,CN=Certificate Templates,CN=Public Key Services,CN=Services,CN=Configuration,DC=daforest,DC=com
+[*]    Vulnerable to: ESC1, ESC2, ESC3_TEMPLATE_2
```

Comment:
Reading through this document with fresh eyes is leaving me wondering why ESC3 is `ESC3_TEMPLATE_2` again. Does the user need to know the difference between `ESC3_TEMPLATE_2` and what I'm guessing is `ESC3_TEMPLATE_1`? Now that we have a doc calling out the different techniques, it leaves one wondering what the difference is.

Reply:
Hmm let me get back to you on this one, this is a good point and I likely need to add more explanation here. Reacted with the eyes emoji so I have a mental note to take a further look into this and edit things up accordingly.

Reply:
Decided the best way to deal with this is to add another section to the overview which breaks down ESC1, ESC2, and ESC3 and notes that ESC3 is similar to ESC2 however it involves two separate certificate templates vs one certificate template and then go into a bit more specifics on that.

Reply:
Tried to clarify this in https://github.com/rapid7/metasploit-framework/pull/17399/commits/5b2a7cc151f71c77e9cedd4e33755bd3f3e72657, please let me know if this works for you or not.